raiden/storage/sqlite.py

```diff
@@ -663,6 +666,63 @@ def update_events(self, events_data: List[Tuple[str, int]]) -> None:
         cursor.executemany("UPDATE state_events SET data=? WHERE identifier=?", events_data)
         self.maybe_commit()
 
+    def get_raiden_events_payment_history_with_timestamps(
+        self,
+        event_types: List[str],
+        limit: int = None,
+        offset: int = None,
+        token_network_address: TokenNetworkAddress = None,
+        partner_address: Address = None,
+    ) -> List[TimestampedEvent]:
+
+        limit, offset = _sanitize_limit_and_offset(limit, offset)
+        cursor = self.conn.cursor()
+        args: List = list(event_types)
+        sql_helper = ",".join("?" * len(event_types))
+
+        if token_network_address and partner_address:
+            query = (
+                "SELECT data, timestamp FROM state_events "
+                "WHERE json_extract(data, '$._type') IN ({}) "
+                "AND json_extract(data, '$.token_network_address') LIKE ? "
+                "AND (json_extract(data, '$.target') LIKE ? OR "
+                "json_extract(data, '$.initiator') LIKE ? ) "
+                "ORDER BY identifier ASC LIMIT ? OFFSET ?"
```

Comment:
Suggestion: The readability of the queries can be improved (at least in the source code), by using a singe string and doing proper indenting:
 
```python
                """
                SELECT 
                  data, timestamp 
                FROM 
                  state_events 
                WHERE 
                  json_extract(data, '$._type') IN ({}) AND 
                  json_extract(data, '$.token_network_address') LIKE ? AND 
                  ( 
                    json_extract(data, '$.target') LIKE ? OR 
                    json_extract(data, '$.initiator') LIKE ? 
                  ) 
                ORDER BY identifier 
                ASC LIMIT ? OFFSET ?
                """
```